src/components/ai-utils.ts

```diff
@@ -2,38 +2,107 @@ import {
   AgentK8sResource,
   ClusterDeploymentK8sResource,
   AgentClusterInstallK8sResource,
+  AgentStatusConditionType,
+  AgentStatusCondition,
+  AgentClusterInstallStatusConditionType,
+  AgentClusterInstallStatusCondition,
 } from './types';
 import { Cluster as AICluster, Host as AIHost } from 'openshift-assisted-ui-lib/dist/src/api';
 
+const conditionsByTypeReducer = (result, condition) => ({ ...result, [condition.type]: condition });
+
+export const getClusterStatus = (
```

Comment:
Such utility functions will be useful for the ACM as well, so they might be worth of adding to the ai-ui-lib. Wdyt?

Reply:
I agree, I'll move it to the ui-lib